vanilla/src/launch/java/org/spongepowered/vanilla/launch/plugin/JavaPluginLoader.java

```diff
@@ -25,37 +25,49 @@
 package org.spongepowered.vanilla.launch.plugin;
 
 import com.google.inject.Injector;
+import org.apache.maven.artifact.versioning.ArtifactVersion;
 import org.spongepowered.common.inject.plugin.PluginModule;
 import org.spongepowered.common.launch.Launch;
 import org.spongepowered.plugin.InvalidPluginException;
 import org.spongepowered.plugin.PluginCandidate;
+import org.spongepowered.plugin.PluginContainer;
 import org.spongepowered.plugin.PluginEnvironment;
 import org.spongepowered.plugin.jvm.JVMPluginContainer;
 import org.spongepowered.plugin.jvm.JVMPluginLoader;
 import org.spongepowered.plugin.jvm.locator.JVMPluginResource;
 
 import java.util.Optional;
 
-public final class JavaPluginLoader extends JVMPluginLoader<JVMPluginContainer> {
+public final class JavaPluginLoader extends JVMPluginLoader<JVMPluginResource, VanillaJavaPluginContainer> {
 
     @Override
-    public Optional<JVMPluginContainer> createPluginContainer(final PluginCandidate<JVMPluginResource> candidate, final PluginEnvironment environment) {
+    public ArtifactVersion version() {
+        return null;
```

Comment:
I didn't plan to actually bump the version of this.

So 1.0 will be sufficient as the version.